src/main/scala/sigmastate/eval/CostingRules.scala

```diff
@@ -479,6 +479,17 @@ trait CostingRules extends SigmaLibrary { IR: RuntimeCosting =>
 
     def isDefined: RCosted[Boolean] = constantSizeProperyAccess(_.isDefined)
     def isEmpty: RCosted[Boolean] = constantSizeProperyAccess(_.isEmpty)
+
+    def map[B](_f: RCosted[T => B]): RCosted[WOption[B]] = {
+      val f = asCostedFunc[T,B](_f)
+      val calcF = f.sliceCalc
+      val costF = f.sliceCost
+      val sizeF = f.sliceSize
+      val v = obj.value.map(calcF)
+      val c = costF(Pair(obj.cost, asSizeOption(obj.size).sizeOpt.get))
```

Comment:
@greenhat please extract common subexpressions like this `asSizeOption(obj.size).sizeOpt`. Otherwise they are executed twice, which is quite costly.

Reply:
Thanks. Fixed.